shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java

```diff
@@ -238,7 +243,56 @@ public ASTNode visitShowWarnings(final ShowWarningsContext ctx) {
     
     @Override
     public ASTNode visitResetStatement(final ResetStatementContext ctx) {
-        return new MySQLResetStatement();
+        MySQLStatementParser.ResetPersistContext persistContext = ctx.resetPersist();
+        if (null != persistContext) {
+            return visit(persistContext);
+        }
+        MySQLResetStatement result = new MySQLResetStatement();
+        for (MySQLStatementParser.ResetOptionContext optCtx : ctx.resetOption()) {
+            ResetOptionSegment opt = (ResetOptionSegment) (visit(optCtx));
+            result.getOptions().add(opt);
+        }
+        return result;
+    }
+
+    @Override
+    public ASTNode visitResetPersist(final MySQLStatementParser.ResetPersistContext ctx) {
+        MySQLResetPersistStatement result = new MySQLResetPersistStatement();
+        if (ctx.existClause() != null) {
+            result.setContainsExistClause(true);
+        }
+        if (ctx.identifier() != null) {
+            result.setIdentifier(new IdentifierValue(ctx.identifier().getText()));
+        }
+        return result;
+    }
+
+    @Override
+    public ASTNode visitResetOption(final MySQLStatementParser.ResetOptionContext ctx) {
+        ResetOptionSegment result = new ResetOptionSegment();
+        result.setStartIndex(ctx.start.getStartIndex());
+        result.setStopIndex(ctx.stop.getStopIndex());
+        if (ctx.MASTER() != null) {
+            result.setTarget(ctx.MASTER().getText());
```

Comment:
@flycash Can we split MySQLResetStatement to MySQLResetMasterStatement and MySQLResetSlaveStatement? They seem to be two statements. At the same time, there should be no need to parse keywords like MASTER, SLAVE, ALL.

Reply:
I think may be we cannot split MySQLResetStatement. Here is the syntax:
```g4
resetStatement
    : RESET resetOption (COMMA_ resetOption)*
    | resetPersist
    ;
```
We can use `RESET` for both `RESET MASTER, SLAVE`, and you can find that I have provided a test case for this scenario. 
Any good idea?

Reply:
Using the same statement is OK, but what I want to say is, do we need to parse out the keywords? These parsed field seem to be useless, whether they can be deleted?

Reply:
I can change them to boolean value like :
```java
boolean master
boolean all
```
Another alternative solution is using two option classes:
```java
class ResetOptionSegement {
}
class ResetMasterOptionSegment{
}
class ResetSlaveOptionSegment{}
```
I think both of these two solutions are ok

Reply:
I use second solution